doc/nrf/protocols/matter/end_product/versioning.rst

```diff
@@ -0,0 +1,77 @@
+.. _ug_dfu_in_matter:
+
+Device Firmaware Update version configuration
+#############################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Device Firmware Update (DFU) is the procedure of upgrading the application firmware version on a device.
+You can transfer the updated images to the device in two ways:
+
+    Upgrade over Bluetooth LE
+
+    `Upgrade over Matter (Matter OTA)<https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/overview/dfu.html>`_, where updates are sent through a wireless connection, like Bluetooth® Low Energy.
+
+Before creating and performing DFU, You need to preper proper versionig configuration.
+Versioning can be done by creating `VERSION file<https://docs.zephyrproject.org/latest/build/version/index.html>`_, or by chenging build configuration in prf.conf.
+
+Versioning with VERSION file
+****************************
+
+VERSION file is resposible for assignment values to MCUBoot version in the following way:
+"MAJOR.MINOR.PATCH.TWEAK"
+
+To use versioning with specified for this file, user needs to create in the application directory a file named VERSION with proper content:
+
+.. code-block:: console
+    VERSION_MAJOR =
+    VERSION_MINOR =
+    PATCHLEVEL =
+    VERSION_TWEAK =
+
+.. note::
+   It is crucial to assign any value to one of the variables, if not, VERSION file will be ignored and Kconfig values will be assgined to version of application.
+
+Versioning by changing configuration
+************************************
+
+.. _ug_matter_dfu_ota:
+
+Device Firmware Upgrade over Matter
+===================================
+
+In case of DFU over Matter user should eddit prj.conf file in the application directory.
+Following lines should be added to mentioned file to change Matter version:
+
+.. code-block:: console
+    CONFIG_CONFIG_CHIP_DEVICE_SOFTWARE_VERSION=<version number>
+
+.. code-block:: console
+    CONFIG_CHIP_DEVICE_SOFTWARE_VERSION_STRING="<version string>"
+
+Additionally Nordic chips uses MCUBoot Image Tool, user should also edit:
+
+.. code-block:: console
+    CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="<app version major number . app version minor number . app patch level . app version tweak>"
+
+.. _ug_matter_dfu_smp:
+
+Device Firmware Upgrade over Bluetooth LE
```

Comment:
It's not only for DFU over BLE. This is for configuring versions of updates images used by the MCUBoot bootloader, but bootloader is involved in any form of DFU. When DFU over Matter is used, one or more MCUBoot update images are bundled into a Matter OTA image that has an extra header and can use a separate versioning scheme.

So when DFU over Matter is used, the versions are checked at two levels:
1. First, a new image is downloaded only if its version is higher than `CHIP_DEVICE_SOFTWARE_VERSION`.
2. Secondly, after downloading an image and rebooting into bootloader, the bootloader checks if the new image version is not less than the current image version (`MCUBOOT_IMGTOOL_SIGN_VERSION`).

The second step is required despite of 1, because 1 doesn't provide any security - Matter images are not signed so the version is used only to check if a new version is available but 2 can enforce that only newer images be installed.

Reply:
nvm, I jumped to this section without reading the previous one :)